sys/kern/tmpfs.c

```diff
@@ -997,10 +998,9 @@ static int tmpfs_resize(tmpfs_mount_t *tfm, tmpfs_node_t *v, size_t newsize) {
 }
 
 static int tmpfs_chtimes(tmpfs_node_t *v, timespec_t *atime, timespec_t *mtime,
-                         cred_t *cred) {
-  int err;
-  if ((err = cred_can_utime(v->tfn_vnode, v->tfn_uid, cred)))
-    return err;
+                         cred_t *cred, va_flags_t vaflags) {
+  if (!cred_can_utime(v->tfn_vnode, v->tfn_uid, cred, vaflags))
```

Comment:
Is there any good reason for this check to be performed by a filesystem instead of code in one layer above, i.e `vfs_*`? I don't expect the code to differ at all between various filesystems...

Reply:
I think it was done to be consistent with previous operations (chmod, and chown). They are also handled on filesystem layer. We can move all these checks to layer above, but I think it sould be done in one separate PR.